Fabric.Authorization.Persistence.SqlServer/Stores/SqlServerGroupStore.cs

```diff
@@ -0,0 +1,118 @@
+
+            return model;
+        }
+
+        public async Task<Group> Get(string id)
+        {
+            var group = await _authorizationDbContext.Groups
+                .Include(g => g.GroupRoles)
+                .ThenInclude(gr => gr.Role)
+                .Include(g => g.GroupUsers)
+                .ThenInclude(gu => gu.User)
+                .SingleOrDefaultAsync(g => g.GroupId.Equals(id, StringComparison.OrdinalIgnoreCase)
+                && !g.IsDeleted);
+
+            if (group == null)
+            {
+                throw new NotFoundException<Group>($"Could not find {typeof(Group).Name} entity with ID {id}");
+            }
+
+            return group.ToModel();
+        }
+
+        public async Task<IEnumerable<Group>> GetAll()
+        {
+            var groups = await _authorizationDbContext.Groups
+                .Include(g => g.GroupRoles)
+                .ThenInclude(gr => gr.Role)
+                .Include(g => g.GroupUsers)
+                .ThenInclude(gu => gu.User)
+                .Where(g => !g.IsDeleted)
+                .ToArrayAsync();
+
+            return groups.Select(g => g.ToModel());
+        }
+
+        public async Task Delete(Group model)
+        {
+            var group = await _authorizationDbContext.Groups
+                .Include(g => g.GroupRoles)
+                .Include(g => g.GroupUsers)
+                .SingleOrDefaultAsync(g => g.GroupId.Equals(model.Id, StringComparison.OrdinalIgnoreCase));
+
+            if (group == null)
+            {
+                throw new NotFoundException<Group>($"Could not find {typeof(Group).Name} entity with ID {model.Id}");
+            }
+
+            group.IsDeleted = true;
+
+            foreach (var groupRole in group.GroupRoles)
+            {
+                groupRole.IsDeleted = true;
+            }
+            foreach (var groupUser in group.GroupUsers)
+            {
+                groupUser.IsDeleted = true;
+            }
+
+            await _authorizationDbContext.SaveChangesAsync();
+        }
+
+        public async Task Update(Group model)
+        {
+            var group = await _authorizationDbContext.Groups
+                .Include(g => g.GroupRoles)
+                .Include(g => g.GroupUsers)
+                .SingleOrDefaultAsync(g => g.GroupId.Equals(model.Id, StringComparison.OrdinalIgnoreCase));
+
+            if (group == null)
+            {
+                throw new NotFoundException<Group>($"Could not find {typeof(Group).Name} entity with ID {model.Id}");
+            }
+
+            model.ToEntity(group);
+
+            _authorizationDbContext.Groups.Update(group);
+            await _authorizationDbContext.SaveChangesAsync();
+        }
+
+        public async Task<bool> Exists(string id)
+        {
+            var group = await _authorizationDbContext.Groups
+                .SingleOrDefaultAsync(g => g.GroupId.Equals(id, StringComparison.OrdinalIgnoreCase)
```

Comment:
Should this be checking the Name property also?

Reply:
Good call. Looking back over the group model I think name is the correct key to use when querying.